### DISCUSSION TITLE
Query in Standard - Report Management in TwinBuilder using PyAEDT

### DISCUSSION THREAD

**Discussion by shabham22** [ansys/pyaedt#5420] [Q&A] [answered]
Hi,

I have written below code for creating Standard report using TwinBuilder pyaedt.
Report is being created, but some operations related to report management is not working:

1. To enable plot Marker either `add_cartesian_x_marker` or `add_cartesian_y_marker` is executing successfully but its not enabling after execution.
2. `export_report_to_jpg` function is not working with `class Standard`, is there any other method to store report jpg.
3. Similarly, `import_traces(input_file, plot_name)` couldn't able to import using csv format.

_**Some which i also tried are working fine like -_ `edit_x_axis_scaling`, `edit_y_axis_scaling`, `add_note`

```
ac_report = tb.post.reports_by_category.standard("mag(VM1.V/AM1.I)")
report1 = ac_report.create(name='Capacitor Impedance Plot')
xscale = ac_report.edit_x_axis_scaling(
    linear_scaling=False,  
    min_scale="0.001kHz", 
    max_scale="10kHz",  
    minor_tick_divs=5,    
    units="kHz"       
)

#Below codes are not wroking
ac_report.add_cartesian_x_marker(100, name="Freq_marker")
ac_report.add_cartesian_y_marker(1e-6, name="Z_marker")
ac_report.export_report_to_jpg(folder_path, "Test")
ac_report.import_traces(input_file, plot_name)
```

**Comment by gmalinve** [comment]
Hi @shabham22,

Thanks for opening this discussion.
Regarding the markers, are you sure you are entering the value for the correct model units you are using? because I tested these two methods myself and they are working. In my case I have a time domain analysis:
![image](https://github.com/user-attachments/assets/583a3c2f-c17e-4bbd-ae4c-020a1f43b84f)
![image](https://github.com/user-attachments/assets/d5754171-8e19-4dd8-bbdd-b97732ebc204)
As you can see in the first screenshot I selected the marker to be at 0.04 s (40 ms in my graph - second screenshot)
Regarding your second issue with the export_report_to_jpg you are calling it in the wrong way. This is a method of post not of the report object (see again my first screenshot).
For the import_traces, what kind of error are you having? because .csv is one of the accepted format, as you can see from one of our available examples: 
https://examples.aedt.docs.pyansys.com/version/dev/examples/low_frequency/team_problem/asymmetric_conductor.html#Create-rectangular-plots-and-import-test-data-into-report.

Hope this helps, kinf regards,

Giulia

**Reply by shabham22** [reply]
Hi @gmalinve,

Thanks for checking the issue and for your detailed reply.

Actually, i want to use the `edit_x_axis_scaling` operation, and it is available with `.post.reports_by_category.standard` only not with `.post.create_report`. So, i tried `add_cartesian_x_marker` and `export_report_to_jpg` with this. Is it possible to use scaling operation with `.post.create_report`?

Also, can you please tell as in example you have used `TR` report, same i was doing with `AC` report, but its not creating AC report. Can you suggest some solution to get AC report or how to analyse AC. As example for AC analysis is not given in help section.

**Comment by gmalinve** [comment, marked as answer]
Hi @shabham22,

Why don't you create your report with the `create_report `method as I showed you in my previous screenshot? this method returns excatly the same object as creating it with `post.reports_by_category.standard` and you can access the `edit_cartesian_marker ` in the same way.
I suggest you use an IDE to debug your code and use the in-built python function `dir()` to discover which properties or methods of an object you can access.
For example, in my case I use PyCharm and if I debug the Standard report object I can see what I can access:
![image](https://github.com/user-attachments/assets/7fc5d40a-bbd6-403a-a01d-c9f8a13870ba).

In my previous example I specified `TR `as my setup sweep name, this is because `TR `is the name of my transient setup so if `AC `is the name of your AC setup you have to specify it in the `create_report`.

Kind regards,

Giulia

**Reply by shabham22** [reply]
Understood. Thanks for reply.

**Reply by shabham22** [reply]
Hi @gmalinve,

I got your point to use `dir()` function to get the objects. It's helpful.
But, How to get the syntax of particular object. Like i want to use object `change_property` 
![image](https://github.com/user-attachments/assets/3169a176-0815-4d8e-9d55-6830e4539db5)

**Reply by Samuelopez-ansys** [reply]
https://aedt.docs.pyansys.com/version/stable/API/Visualization.html
